### PR TITLE
Fix client transaction support

### PR DESCRIPTION
**Pull request by dain**

### CODE REVIEW

**Comment by martint** [comment]
Needs a better commit message. What does it fix?

File: presto-main/src/test/java/com/facebook/presto/server/TestServer.java [hunk 1]
```diff
@@ -130,12 +135,64 @@ public void testQuery()
...
+            }
+
+            if (queryResults.getNextUri() == null) {
+                break;
+            }
+            queryResults = client.execute(prepareGet().setUri(queryResults.getNextUri()).build(), createJsonResponseHandler(jsonCodec(QueryResults.class)));
+        }
+        assertNull(queryResults.getError());
+
+        // only the system catalog exists by default
```
Comment (electrum) [inline presto-main/src/test/java/com/facebook/presto/server/TestServer.java]: This comment doesn't make sense

File: presto-main/src/test/java/com/facebook/presto/server/TestServer.java [hunk 2]
```diff
@@ -130,12 +135,64 @@ public void testQuery()
...
+        List<List<Object>> rows = data.build();
+        assertEquals(rows, ImmutableList.of(ImmutableList.of(TRUE)));
```
Comment (electrum) [inline presto-main/src/test/java/com/facebook/presto/server/TestServer.java]: We don't actually care about the data for this. It's a dummy row to make some clients happy. Instead, we need to verify that we got the `PRESTO_STARTED_TRANSACTION_ID` header.

File: presto-main/src/test/java/com/facebook/presto/server/TestServer.java [hunk 3]
```diff
@@ -130,12 +137,61 @@ public void testQuery()
...
+                .setHeader(PRESTO_SOURCE, "source")
+                .build();
+
+        QueryResults queryResults = client.execute(request, createJsonResponseHandler(jsonCodec(QueryResults.class)));
+        while (queryResults.getNextUri() != null) {
+            queryResults = client.execute(prepareGet().setUri(queryResults.getNextUri()).build(), createJsonResponseHandler(jsonCodec(QueryResults.class)));
+        }
+
+        assertNotNull(queryResults.getError());
+        assertEquals(queryResults.getError().getErrorCode(), StandardErrorCode.INCOMPATIBLE_CLIENT.toErrorCode().getCode());
```
Comment (electrum) [inline presto-main/src/test/java/com/facebook/presto/server/TestServer.java]: Static import the error code

**Pull request closed without merging**